### PR TITLE
Adding commands for common Koji tasks

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Designed to replace this script with a more automated version -- https://github.com/Katello/katello-misc/blob/master/rel-eng/koji_create_buildroots.sh

**Comment by ehelms** [comment]
Note that this will run in 'dry run' mode by default and output a preview of all commands so it is safe to run them and analyze the output for flaws (--commit will run them live so don't use that option unless you truly want to test it against Koji).

File: lib/tool_belt/commands/koji_command.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            signal_usage_error "OS must be one or more of #{operating_systems}" unless (os - operating_systems).empty?
+            os
+          end
+        end
+      end
+
+      module KatelloVersionOption
+        def self.included(base)
+          base.send :option, "--katello-version", "KATELLO_VERSION", "Katello version to build client tags for" do |version|
+            versions = %w(nightly 2.1 2.2 2.3)
```
Comment (jlsherrill) [inline lib/tool_belt/commands/koji_command.rb]: might be nice to make this config option?

Comment (ehelms) [inline lib/tool_belt/commands/koji_command.rb]: I could just remove this validation or setup a configuration file to read from

**Comment by jlsherrill** [comment]
Client issue i discovered, it does not seem to be creating the  katello-VERSION-OS tags

File: lib/tool_belt/koji_builder.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        # a version and then not again unless done as a manual command outside
+        # of this function
+        if execute("list-tags katello-#{@katello_version}-#{os}").first.nil?
+          execute("clone-tag katello-nightly-#{os} katello-#{@katello_version}-#{os}")
+          execute("clone-tag katello-thirdparty-#{os} katello-#{@katello_version}-thirdparty-#{os}")
+        end
+
+        execute("add-tag katello-#{@katello_version}-#{os}-build  --arches=x86_64")
+
+        if SERVER_OSES.include?(os)
```
Comment (ehelms) [inline lib/tool_belt/koji_builder.rb]: Add this wrapper here for building non-server OS tags such as katello-nightly-fedora23

File: lib/tool_belt/koji_builder.rb [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    attr_reader :version, :systools
+
+    def initialize(args = {})
+      @katello_version = args.fetch(:katello_version, 'nightly') || 'nightly'
+      @systools = args.fetch(:systools) || SysTools.new
+      @foreman_version = VERSIONS[@katello_version]
+    end
+
+    def build_katello_tags(os)
+      operating_systems = os.nil? ? CLIENT_OSES : os
```
Comment (ehelms) [inline lib/tool_belt/koji_builder.rb]: Switched this to all OSes essentially for building non-server OS tags such as katello-nightly-fedora23

**Comment by jlsherrill** [comment]
ACK this worked great for me for 3.1

**Pull request merged**